File: p2b/src/jungle/pieces/lion.py
"""
表示丛林游戏中的狮子棋子。
狮子可以水平和垂直跳跃越过水方格。
"""

from typing import TYPE_CHECKING
from .piece import Piece

if TYPE_CHECKING:
    from ..player import Player
    from ..squares.square import Square


class Lion(Piece):
    """
    丛林游戏中的狮子棋子。
    
    狮子的等级为7，可以水平和垂直跳跃越过水方格。
    """
    
    def __init__(self, owner: 'Player', square: 'Square'):
        """
        使用拥有者和方格构造一个狮子。
        狮子的等级为7。
        
        Args:
            owner: 拥有这个狮子的玩家
            square: 这个狮子占据的方格
        """
        super().__init__(owner, square, 7)
    
    def can_leap_horizontally(self) -> bool:
        """
        指示狮子可以水平跳跃。
        
        Returns:
            True
        """
        return True
    
    def can_leap_vertically(self) -> bool:
        """
        指示狮子可以垂直跳跃。
        
        Returns:
            True
        """
        return True

File: p2a/src/jungle/pieces/Lion.java
package jungle.pieces;

import jungle.Player;
import jungle.squares.Square;

/**
 * 表示丛林游戏中的狮子棋子。
 * 狮子可以水平和垂直跳跃越过水方格。
 */
public class Lion extends Piece {

    /**
     * 使用拥有者和方格构造一个狮子。
     * 狮子的等级为7。
     * 
     * @param owner     拥有这个狮子的玩家
     * @param square    这个狮子占据的方格
     */
    public Lion(Player owner, Square square) {
        super(owner, square, 7);
    }

    /**
     * 指示狮子可以水平跳跃。
     * 
     * @return true
     */
    @Override
    public boolean canLeapHorizontally() {
        return true;
    }

    /**
     * 指示狮子可以垂直跳跃。
     * 
     * @return true
     */
    @Override
    public boolean canLeapVertically() {
        return true;
    }
}
